### PR TITLE
Facility view revamp

### PR DESCRIPTION
**Pull request by jasonrig**
Improved load speed with only experiment/dataset listings loaded initially and datafile listings loaded on-demand.
"Load more" button only loads the next set of datasets, rather than reloading the whole list.

(pycharm made a lot of html/js formatting adjustments)

### CODE REVIEW

File: tardis/tardis_portal/static/js/facility_view.js [hunk 1]
```diff
@@ -1,265 +1,313 @@
...
+    var datasetDetailRes = $resource('/facility/fetch_datafiles/:dataset_id/',
+        {}, {
+            'get': {method: 'GET', isArray: true}
+        });
+    var facilityListRes = $resource('/facility/fetch_facilities_list/', {}, {
+        'get': {method: 'GET', isArray: true}
+    });
+    var facilityDataRes = $resource('/facility/fetch_data/:facility_id/:start_index/:end_index/', {
+        start_index: 0,
+        end_index: 50
```
Comment (grischa) [inline tardis/tardis_portal/static/js/facility_view.js]: Codacy Issue found: [Identifier 'end_index' is not in camel case.](https://www.codacy.com/app/mytardis/mytardis/file/1288071727/issues/source?bid=2125231&fileBranchId=2946418#l50)

File: tardis/tardis_portal/static/js/facility_view.js [hunk 2]
```diff
@@ -1,265 +1,313 @@
...
+    var countRes = $resource('/facility/fetch_data/:facility_id/count/');
+    var datasetDetailRes = $resource('/facility/fetch_datafiles/:dataset_id/',
+        {}, {
+            'get': {method: 'GET', isArray: true}
+        });
+    var facilityListRes = $resource('/facility/fetch_facilities_list/', {}, {
+        'get': {method: 'GET', isArray: true}
+    });
+    var facilityDataRes = $resource('/facility/fetch_data/:facility_id/:start_index/:end_index/', {
+        start_index: 0,
```
Comment (grischa) [inline tardis/tardis_portal/static/js/facility_view.js]: Codacy Issue found: [Identifier 'start_index' is not in camel case.](https://www.codacy.com/app/mytardis/mytardis/file/1288071727/issues/source?bid=2125231&fileBranchId=2946418#l49)

**Comment by jasonrig** [comment]
Some refactoring; untested, so don't merge yet.

File: tardis/tardis_portal/static/js/facility_view.js (hunk 1 above)
Comment (grischa) [inline tardis/tardis_portal/static/js/facility_view.js]: Codacy Issue found: [Identifier 'end_index' is not in camel case.](https://www.codacy.com/app/mytardis/mytardis/file/1314039923/issues/source?bid=2125231&fileBranchId=2946418#l50)

File: tardis/tardis_portal/static/js/facility_view.js (hunk 2 above)
Comment (grischa) [inline tardis/tardis_portal/static/js/facility_view.js]: Codacy Issue found: [Identifier 'start_index' is not in camel case.](https://www.codacy.com/app/mytardis/mytardis/file/1314039923/issues/source?bid=2125231&fileBranchId=2946418#l49)

**Comment by jasonrig** [comment]
Further performance enhancements expected when PR for my branch [size-string-to-bigint](https://github.com/jasonrig/mytardis/tree/size-string-to-bigint) is submitted. This includes a major migration to convert DataFile's size field from CharField to BigInteger, thus allowing django's aggregation feature to do the dataset size summing in an efficient way.

**Pull request closed without merging**